dataplane/linux/bpf_ep_mgr.go

```diff
@@ -329,29 +263,9 @@ func (m *bpfEndpointManager) onWorkloadEnpdointRemove(msg *proto.WorkloadEndpoin
 	wlID := *msg.Id
 	log.WithField("id", wlID).Debug("Workload endpoint removed")
 	oldWEP := m.allWEPs[wlID]
-	for _, t := range oldWEP.Tiers {
-		for _, pol := range t.IngressPolicies {
-			polSet := m.policiesToWorkloads[proto.PolicyID{
-				Tier: t.Name,
-				Name: pol,
-			}]
-			if polSet == nil {
-				continue
-			}
-			polSet.Discard(wlID)
-		}
-		for _, pol := range t.EgressPolicies {
-			polSet := m.policiesToWorkloads[proto.PolicyID{
-				Tier: t.Name,
-				Name: pol,
-			}]
-			if polSet == nil {
-				continue
-			}
-			polSet.Discard(wlID)
-		}
-	}
```

Comment:
Bug was that this code didn't match the code above: it was missing the cleanup of profiles.  To fix, I made the code common and called the same code from both places.